jgrapht-core/src/main/java/org/jgrapht/generate/RandomRegularGraphGenerator.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * (C) Copyright 2018, by Emilio Cruciani and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+
+package org.jgrapht.generate;
+
+import org.jgrapht.Graph;
+import org.jgrapht.VertexFactory;
+
+import java.util.*;
+
+/**
+ * Generate a random $d$-regular undirected graph with $n$ vertices.
+ * A regular graph is a graph where each vertex has the same degree, i.e. the same number of neighbors.
+ *
+ * <p>
+ * The algorithm for the simple case, proposed in [SW99] and extending the one for the non-simple case,
+ * runs in expected $\mathcal{O}(nd^2)$ time.
+ * It has been proved in [KV03] to sample from the space of random d-regular graphs
+ * in a way which is asymptotically uniform at random when $d = \mathcal{O}(n^{1/3 - \epsilon})$.
+ *
+ * <p>
+ * [KV03] Kim, Jeong Han, and Van H. Vu.
+ * "Generating random regular graphs."
+ * Proceedings of the thirty-fifth annual ACM symposium on Theory of computing. ACM, 2003.
+ *
+ * [SW99] Steger, Angelika, and Nicholas C. Wormald.
+ * "Generating random regular graphs quickly."
+ * Combinatorics, Probability and Computing 8.4 (1999): 377-396.
+ *
+ * @author Emilio Cruciani
+ * @since March 2018
+ *
+ * @param <V> graph node type
+ * @param <E> graph edge type
+ */
+public class RandomRegularGraphGenerator<V, E> implements GraphGenerator<V, E, V> {
+
+    private final int n;
+    private final int d;
+
+    /**
+     * Construct a new RandomRegularGraphGenerator.
+     *
+     * @param n number of nodes
+     * @param d degree of nodes
+     * @throws IllegalArgumentException if number of nodes is negative
+     * @throws IllegalArgumentException if degree is negative
+     * @throws IllegalArgumentException if degree is greater than number of nodes
+     * @throws IllegalArgumentException if the value "n * k" is odd
+     */
+    public RandomRegularGraphGenerator(int n, int d) {
+        if (n < 0) {
+            throw new IllegalArgumentException("number of nodes must be non-negative");
+        }
+        if (d < 0) {
+            throw new IllegalArgumentException("degree of nodes must be non-negative");
+        }
+        if (d > n) {
+            throw new IllegalArgumentException("degree of nodes must be smaller than or equal to number of nodes");
+        }
+        if ((n * d) % 2 != 0) {
+            throw new IllegalArgumentException("value 'n * k' must be even");
```

Comment:
Minor nit: I believe this should say `'n * d'` instead of `'n * k'`. :)

Reply:
Thanks for that. Fixed it in the Javadocs too ;-)